### PR TITLE
remove file locking

### PR DESCRIPTION
**Pull request by bgruening**
This seems to break NFS setups.

https://github.com/usegalaxy-eu/issues/issues/552

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by bgruening** [comment]
Can I have a santiy check from a real developer @davelopez ?

File: lib/galaxy/jobs/runners/util/fork_safe_write.py [hunk 1]
```diff
@@ -13,6 +13,6 @@ def fork_safe_write(path: str, contents: str):
     # file is open for writing.
     with open(path, "w", encoding="utf-8") as f:
         f.write(unicodify(contents))
-        fcntl.flock(f.fileno(), fcntl.LOCK_EX)
-    with open(path) as f:
-        fcntl.flock(f.fileno(), fcntl.LOCK_SH)
+        #fcntl.flock(f.fileno(), fcntl.LOCK_EX)
```
Comment (sanjaysrikakulam) [inline lib/galaxy/jobs/runners/util/fork_safe_write.py:16]: Don't we need an exclusive `f.flush()` call?

Comment (bgruening) [reply lib/galaxy/jobs/runners/util/fork_safe_write.py:16]: `with` should close the file handle and flush imho

Comment (sanjaysrikakulam) [reply lib/galaxy/jobs/runners/util/fork_safe_write.py:16]: Closes, yes; flush, I am not so sure. I have always flushed exclusively. Let's see what @davelopez says.

Comment (davelopez) [reply lib/galaxy/jobs/runners/util/fork_safe_write.py:16]: It should be enough to use the context manager :+1:

**Pull request merged**